lightning/src/ln/channelmanager.rs

```diff
@@ -13647,15 +13606,32 @@ where
 							&& funding_txo == pending_funding
 							&& channel_id == pending_chan_id;
 						if for_same_channel {
+							debug_assert!(update.update_id >= pending_update.update_id);
 							if pending_update.updates.iter().any(|upd| matches!(upd, ChannelMonitorUpdateStep::ChannelForceClosed { .. })) {
 								// If the background event we're looking at is just
 								// force-closing the channel which already has a pending
 								// force-close update, no need to duplicate it.
 								continue 'each_bg_event;
 							}
+							update.update_id = pending_update.update_id.saturating_add(1);
+							updated_id = true;
 						}
 					}
 				}
+				let mut per_peer_state =  per_peer_state.get(counterparty_node_id)
+					.expect("If we have pending updates for a channel it must have an entry")
+					.lock().unwrap();
+				if updated_id {
+					per_peer_state
+						.closed_channel_monitor_update_ids.entry(*channel_id)
+						.and_modify(|v| *v = cmp::max(update.update_id, *v))
+						.or_insert(update.update_id);
+				}
```

Comment:
This change feels like it should've been part of 64895503afe49b60cdfcadc63cdcfe202c7b2dc6